src/node/build/bundle.ts

```diff
@@ -98,9 +99,12 @@ export async function bundle(
           app: path.resolve(APP_PATH, ssr ? 'ssr.js' : 'index.js'),
           ...input
         },
-        // important so that each page chunk and the index export things for each
-        // other
-        preserveEntrySignatures: 'allow-extension',
+        // @ts-ignore skip setting it for rolldown-vite since it doesn't support `preserveEntrySignatures` yet
+        ...(vite.rolldownVersion
+          ? undefined
+          : // important so that each page chunk and the index export things for each
+            // other
+            { preserveEntrySignatures: 'allow-extension' }),
```

Comment:
This part can be reverted, once https://github.com/rolldown/rolldown/issues/3500 lands.